client/pages/verify.js

```diff
@@ -0,0 +1,228 @@
+// Module imports
+import React from 'react'
+
+
+
+
+
+// Component imports
+import { actions, connect, actionStatus } from '../store'
+import { Router, Link } from '../routes'
+import PageWrapper from '../components/PageWrapper'
+import httpStatus from '../helpers/httpStatus'
+import PasswordField from '../components/PasswordField'
+import { passwordPattern } from '../data/RegExpr'
+
+
+
+
+
+@connect
+class Verify extends React.Component {
+  /***************************************************************************\
+    Class Properties
+  \***************************************************************************/
+
+  state = {
+    password: '',
+    submitted: false,
+    submitting: false,
+  }
+
+
+
+
+
+  /***************************************************************************\
+    Private Methods
+  \***************************************************************************/
+
+  _handleSubmit = async (event) => {
+    const {
+      password,
+      token,
+    } = this.state
+
+    event.preventDefault()
+
+    this.setState({ submitting: true })
+
+    const resetResponse = await this.props.resetPassword({
+      password,
+      token,
+    })
+
+    if (resetResponse.status === actionStatus.SUCCESS) {
+      this.setState({
+        submitted: true,
+        submitting: false,
+      })
+      Router.pushRoute('home', { authenticate: true })
+    }
+  }
+
+
+
+
+
+  /***************************************************************************\
+    Public Methods
+  \***************************************************************************/
+
+  static async getInitialProps ({ query, store, res }) {
+    // eslint-disable-next-line id-length
+    const { type, t: token } = query
+    let destination = null
+    let response = null
+
+    switch (type) {
+      case 'reset':
+        response = await actions.verifyResetToken(token)(store.dispatch)
+        if (response.status === actionStatus.SUCCESS) {
+          destination = null
+        }
+        break
+      case 'email':
+        response = await actions.verifyEmailToken(token)(store.dispatch)
+        if (response.status === actionStatus.SUCCESS) {
+          destination = '/profile'
+        }
+        break
+      case 'session':
+        response = await actions.verifySessionToken(token)(store.dispatch)
+        if (response.status === actionStatus.SUCCESS) {
+          destination = '/?authenticate=true'
+        }
+        break
+      default:
+        destination = '/'
+        break
+    }
+    if (destination) {
+      if (res) {
+        res.writeHead(httpStatus.FOUND, {
+          Location: `${destination}`,
```

Comment:
This string interpolation isn't really needed.

```suggestion
          Location: destination,
```